CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+cmake_minimum_required(VERSION 3.6.0)
+
+project(djinni_support_lib)
+
+include(GNUInstallDirs)
+
+set(SRC_SHARED
+  "djinni/djinni_common.hpp"
+  "djinni/proxy_cache_interface.hpp"
+  "djinni/proxy_cache_impl.hpp"
+)
+
+set(SRC_JNI
+  "djinni/jni/djinni_support.hpp"
+  "djinni/jni/Marshal.hpp"
+  "djinni/jni/djinni_support.cpp"
+)
+
+set(SRC_OBJC
+  "djinni/objc/DJICppWrapperCache+Private.h"
+  "djinni/objc/DJIError.h"
+  "djinni/objc/DJIMarshal+Private.h"
+  "djinni/objc/DJIObjcWrapperCache+Private.h"
+  "djinni/objc/DJIError.mm"
+  "djinni/objc/DJIProxyCaches.mm"
+)
+
+option(DJINNI_STATIC_LIB "Build Djinni support library as a static library instead of dynamic (the default)." off)
+if(DJINNI_STATIC_LIB)
+  add_library(djinni_support_lib STATIC ${SRC_SHARED})
+else()
+  add_library(djinni_support_lib SHARED ${SRC_SHARED})
+endif()
+source_group("" FILES ${SRC_SHARED})
+
+set_target_properties(djinni_support_lib PROPERTIES
+  CXX_STANDARD 11
+  CXX_STANDARD_REQUIRED true
+  CXX_EXTENSIONS false
+)
+
+target_include_directories(djinni_support_lib PUBLIC
+ # $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/djinni>
+  #$<INSTALL_INTERFACE:include/djinni>  # TODO, needs extra test, and what about Windows
```

Comment:
why are these lines commented out? I don't like merging TODOs and commented out code. 😕
Can you maybe add a link to an issue that is supposed to resolve the todo?

Reply:
please see my reply to the next comment

Reply:
I still don't like having this TODO in the code as I cannot understand what the comment means. Can we agree on you removing these lines if they are not required right now? Or, as I mentioned above, could you stick a link to an issue to it or add more details to the comment? 

Sry for being so stubborn about this, if you feel different about this pls tell me! I feel like even in this early stage it's not a waste of time to talk about code quality expectations, instead of silently accepting what one doesn't like! :)

Reply:
sure, I can remove it. 
I like todos in the code, since they are close and clear. A ticket tells me nothing when I am in the editor.
But maybe we should have a multi staging branching strategy, TODOs in the dev branch but not in main or so,  we discuss this extra in an other context.

About the concrete TODO, what I mean, on uinx like OSes, it's easy, you know where the include files go, usually
On Windows .... not sure about `INSTALL_INTERFACE/include` ...